jobserver/views/index.py

```diff
@@ -8,15 +8,55 @@ class Index(TemplateView):
 
     def get_context_data(self, **kwargs):
         job_requests = JobRequest.objects.select_related(
-            "workspace", "workspace__project", "workspace__project__org"
+            "created_by", "workspace", "workspace__project", "workspace__project__org"
+        ).order_by("-created_at")
+
+        all_job_requests = job_requests[:10]
+
+        if not self.request.user.is_authenticated:
+            return super().get_context_data(**kwargs) | {
+                "all_job_requests": all_job_requests,
+            }
+
+        analysis_requests = self.request.user.analysis_requests.select_related(
+            "project", "project__org"
         ).order_by("-created_at")[:5]
+
+        applications = self.request.user.applications.order_by("-created_at")[:5]
+
+        projects = [
+            {
+                "name": m.project.title,
+                "url": m.project.get_absolute_url(),
+            }
+            for m in self.request.user.project_memberships.select_related(
+                "project", "project__org"
+            ).order_by("-created_at")[:5]
+        ]
+
+        user_job_requests = job_requests.filter(created_by=self.request.user)[:5]
+
         workspaces = (
-            Workspace.objects.filter(is_archived=False)
-            .select_related("project", "project__org", "repo")
-            .order_by("name")
+            Workspace.objects.filter(
+                is_archived=False, project__in=self.request.user.projects.all()
+            )
+            .select_related("project", "project__org")
+            .order_by("-created_at")
         )
 
+        counts = {
+            "applications": self.request.user.applications.count(),
+            "job_requests": self.request.user.job_requests.count(),
+            "projects": self.request.user.project_memberships.count(),
+            "workspaces": workspaces.count(),
+        }
+
         return super().get_context_data(**kwargs) | {
-            "job_requests": job_requests,
-            "workspaces": workspaces,
+            "all_job_requests": all_job_requests,
+            "analysis_requests": analysis_requests,
+            "applications": applications,
+            "counts": counts,
+            "job_requests": user_job_requests,
+            "projects": projects,
+            "workspaces": workspaces[:5],
```

Comment:
This is a little tricy because `projects` is a new structure created from the user's memberships (which is where we do the limiting) but could work out.

Reply:
Good point. I've just had a go at testing it locally and it appeared to work, or rather, the unit tests didn't break. I also tried removing the `all_job_requests` variable, which seemed okay, but means you could return back a different number of job requests depending on whether you're logged in or not and I don't know if that's a good thing.

It's not important, it just felt like it might be possible to tidy up the view a bit

Reply:
Oooh nice catch on the `all_job_requests`, fixed.

I've moved all the limits to the context dict, I think you're right that they're better there despite the projects one having to live elsewhere, net win 👍